docker-entrypoint.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
+
+# A beginning user should be able to docker run image bash (or sh) without
+# needing to learn about --entrypoint
+# https://github.com/docker-library/official-images#consistency
+
+set -e
+
+# run command if it is not starting with a "-", is not a git subcommand and is an executable in PATH
+if [ "$#" -gt 0 -a "${1#-}" == "$1" -a ! -x "/usr/libexec/git-core/git-$1" ] && which "$1" > /dev/null 2>&1 ; then
```

Comment:
For consistencies sake (and shell best practices) I'd curly-brace each var.
```suggestion
if [ "${#}" -gt 0 -a "${1#-}" == "${1}" -a ! -x "/usr/libexec/git-core/git-${1}" ] && which "${1}" > /dev/null 2>&1 ; then
```